src/atoms/TextInput/AnimatedLabel.native.js

```diff
@@ -0,0 +1,211 @@
+  line-height: ${styles.text.lineHeight};
+  color: ${styles.text.color};
+`;
+
+const onFocus = ({ AnimationConfig, labelAnimatedValue }) => {
+  Animated.timing(labelAnimatedValue, {
+    toValue: AnimationConfig.FINAL_ANIMATION_VALUE,
+    duration: AnimationConfig.ANIMATION_DURATION,
+  }).start();
+};
+
+const onBlur = ({ AnimationConfig, labelAnimatedValue }) => {
+  Animated.timing(labelAnimatedValue, {
+    toValue: AnimationConfig.INITIAL_ANIMATION_VALUE,
+    duration: AnimationConfig.ANIMATION_DURATION,
+  }).start();
+};
+
+const getColorInterpolation = ({ AnimationConfig, labelAnimatedValue }) => {
+  return labelAnimatedValue.interpolate({
+    inputRange: [AnimationConfig.INITIAL_ANIMATION_VALUE, AnimationConfig.FINAL_ANIMATION_VALUE],
+    outputRange: [AnimationConfig.INITIAL_LABEL_COLOR, AnimationConfig.FINAL_LABEL_COLOR],
+  });
+};
+
+const getFontInterpolation = ({ AnimationConfig, labelAnimatedValue, hasText }) => {
+  if (hasText) return AnimationConfig.FINAL_FONT_SIZE;
+
+  return labelAnimatedValue.interpolate({
+    inputRange: [AnimationConfig.INITIAL_ANIMATION_VALUE, AnimationConfig.FINAL_ANIMATION_VALUE],
+    outputRange: [AnimationConfig.INITIAL_FONT_SIZE, AnimationConfig.FINAL_FONT_SIZE],
+  });
+};
+
+const getTopInterpolation = ({ AnimationConfig, labelAnimatedValue, hasText }) => {
+  if (hasText) return AnimationConfig.FINAL_TOP_POSITION;
+
+  return labelAnimatedValue.interpolate({
+    inputRange: [AnimationConfig.INITIAL_ANIMATION_VALUE, AnimationConfig.FINAL_ANIMATION_VALUE],
+    outputRange: [AnimationConfig.INITIAL_TOP_POSITION, AnimationConfig.FINAL_TOP_POSITION],
+  });
+};
+
+const getLeftInterpolation = ({ AnimationConfig, labelAnimatedValue, hasText }) => {
+  if (hasText) return AnimationConfig.FINAL_LEFT_POSITION;
+
+  return labelAnimatedValue.interpolate({
+    inputRange: [AnimationConfig.INITIAL_ANIMATION_VALUE, AnimationConfig.FINAL_ANIMATION_VALUE],
+    outputRange: [AnimationConfig.INITIAL_LEFT_POSITION, AnimationConfig.FINAL_LEFT_POSITION],
+  });
+};
+
+const getInitialTopPosition = ({ layoutDimensions, variant }) => {
+  if (variant === 'outlined') {
+    if (IS_ANDROID) {
+      return layoutDimensions.height / ANDROID_OUTLINED_INITIAL_TOP_DIVISOR;
+    } else {
+      return layoutDimensions.height / IOS_OUTLINED_INITIAL_TOP_DIVISOR;
+    }
+  } else if (IS_ANDROID) {
+    return layoutDimensions.height / ANDROID_FILLED_INITIAL_TOP_DIVISOR;
+  } else {
+    return layoutDimensions.height / IOS_FILLED_INITIAL_TOP_DIVISOR;
+  }
+};
+
+const getInitialLeftPosition = ({ layoutDimensions }) => layoutDimensions.x;
+
+const getFinalLabelColor = ({ theme, hasError }) =>
+  hasError ? getColor(theme, 'shade.960') : getColor(theme, 'primary.900');
+
+const AnimatedLabel = ({
+  isFocused,
+  children,
+  hasText,
+  disabled,
+  layoutDimensions,
+  variant,
+  hasError,
+}) => {
+  const theme = useContext(ThemeContext);
+
+  const AnimationConfig = {
+    ANIMATION_DURATION: 100,
```

Comment:
These are not constants so make all of them `camelCased`

Reply:
All of these are constants. Once defined here, their values will not change unless props have changed. I'm trying to keep this a standard practice for animations, ```AnimationConfig.SOME_ANIMATION_VALUE```. 

This is supposed to be the object that is the source of truth for all our animation values for that component.

Reply:
@kamleshchandnani  Does something like this make more sense?
```
const AnimationConfig = {
    fontSize: {
      initial: parseInt(theme.fonts.size.medium, 10),
      final: parseInt(theme.fonts.size.xsmall, 10),
    },
    topPosition: {
      initial: getInitialTopPosition({ layoutDimensions, variant }),
      final: 0,
    },
    leftPosition: {
      initial: getInitialLeftPosition({ layoutDimensions }),
      final: 0,
    },
    labelColor: {
      initial: getColor(theme, 'shade.960'),
      final: getFinalLabelColor({ theme, hasError }),
    },
    animationValue: {
      initial: 0,
      final: 1,
    },
    duration: 100,
  };
```